cloudsync_onedrive.py

```diff
@@ -381,44 +386,70 @@ def _direct_api(self, action, path=None, *, url=None, stream=None, data=None, he
 
         return res
 
+    def _safe_direct_api(self, path, method="get", default=None):
+        # wrapper for _direct_api that raises only connectivity errors - other errors are caught and logged
+        try:
+            return self._direct_api(method, path)
+        except (CloudDisconnectedError, CloudTokenError, CloudTemporaryError):
+            raise
+        except Exception as e:
+            log.warning("%s failed with %s", path, repr(e))
```

Comment:
i'm curious what specific exceptions should return the default i/e what behavior we think means "default is needed"

Reply:
Essentially what I was going for: errors that are likely to be fixed by retrying the operation are re-raised (token expired, internet connection went down) and everything else is suppressed. Specific examples of the latter: KeyError for the missing attribute, CloudFileNotFoundError for the sites-search-404 issue.